### DISCUSSION TITLE
What units?

### DISCUSSION THREAD

**Discussion by nickrout** [alandtse/tesla#228] [Q&A]
I am loving this integration, and loving my Tesla too.

Just a small query, in the attributes of the Location Tracker, what is the unit of speed? mph? km/h? european swallow?

**Comment by alandtse** [comment]
Unladen African Swallow.

It should be tied to the UI settings of the vehicle I believe. https://tesla-api.timdorr.com/vehicle/state/drivestate#response

**Reply by nickrout** [reply]
Thank you.

**Reply by nickrout** [reply]
We went for a journey today and I believe it reports in mph, even though our car screen reports in km/hr. Easy conversion. 

Weirdly range reports in the km/hr, charging reports in km/hr, vehicle mileage reports in km, but the vehicle's odometer (an attribute of vehicle_state_sensor) reports in miles.

**Reply by alandtse** [reply]
Oh interesting. Thanks for reporting. I may have misremembered, but I thought all the data was coming based on the UI settings. Perhaps it is defaulting to US units and we'll just need to convert on our end.

**Reply by nickrout** [reply]
By the way I am in New Zealand, where we expect metric units. Temps are all in deg C, so all good there.

If you want any more info, do let me know.
